packages/pds/src/auth-verifier.ts

```diff
@@ -264,25 +281,71 @@ export class AuthVerifier {
   }
 
   moderator = async (
-    reqCtx: ReqCtx,
+    ctx: ReqCtx,
   ): Promise<AdminTokenOutput | ModServiceOutput> => {
-    if (isBearerToken(reqCtx.req)) {
-      return this.modService(reqCtx)
+    if (isBearerToken(ctx.req)) {
+      return this.modService(ctx)
     } else {
-      return this.adminToken(reqCtx)
+      return this.adminToken(ctx)
+    }
+  }
+
+  protected async validateAdminToken({
+    req,
+  }: ReqCtx): Promise<AdminTokenOutput> {
+    const parsed = parseBasicAuth(req.headers.authorization)
+    if (!parsed) {
+      throw new AuthRequiredError()
+    }
+    const { username, password } = parsed
+    if (username !== 'admin' || password !== this._adminPass) {
+      throw new AuthRequiredError()
     }
+
+    return { credentials: { type: 'admin_token' } }
   }
 
-  async validateBearerToken(
-    req: express.Request,
+  protected async validateRefreshToken(
+    ctx: ReqCtx,
+    verifyOptions?: Omit<jose.JWTVerifyOptions, 'audience'>,
+  ): Promise<ValidatedRefreshBearer> {
+    const result = await this.validateBearerToken(ctx, [AuthScope.Refresh], {
+      ...verifyOptions,
+      // when using entryway, proxying refresh credentials
+      audience: this.dids.entryway ? this.dids.entryway : this.dids.pds,
+    })
+    const tokenId = result.payload.jti
+    if (!tokenId) {
+      throw new AuthRequiredError(
+        'Unexpected missing refresh token id',
+        'MissingTokenId',
+      )
+    }
+    return { ...result, tokenId }
+  }
+
+  protected async validateBearerToken(
+    ctx: ReqCtx,
     scopes: AuthScope[],
     verifyOptions?: jose.JWTVerifyOptions,
   ): Promise<ValidatedBearer> {
-    const token = bearerTokenFromReq(req)
+    this.setAuthHeaders(ctx)
+
+    const token = bearerTokenFromReq(ctx.req)
     if (!token) {
       throw new AuthRequiredError(undefined, 'AuthMissing')
     }
-    const payload = await verifyJwt({ key: this._jwtKey, token, verifyOptions })
+
+    const { payload, protectedHeader } = await this.jwtVerify(
+      token,
+      verifyOptions,
+    )
+
+    if (protectedHeader.typ) {
+      // Only OAuth Provider sets this claim
+      throw new InvalidRequestError('Malformed token', 'InvalidToken')
+    }
```

Comment:
The reason I did this is because DPoP states that we must make sure that we will not allow DPoP Proofs (`typ: "dpop+jwt"`) in places where they shouldn't be allowed.

@devinivy Please see the updated comment here 5b5e754d1da852573b9fba42357c667c188e3e9f